src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java

```diff
@@ -521,35 +544,35 @@ void testCopyingExistingWithARestartMidwayThrough() {
               put(MongoSourceConfig.COLLECTION_CONFIG, coll.getNamespace().getCollectionName());
               put(MongoSourceConfig.STARTUP_MODE_CONFIG, StartupMode.COPY_EXISTING.propertyValue());
               put(MongoSourceConfig.POLL_MAX_BATCH_SIZE_CONFIG, "25");
-              put(MongoSourceConfig.POLL_AWAIT_TIME_MS_CONFIG, "2000");
+              put(MongoSourceConfig.POLL_AWAIT_TIME_MS_CONFIG, "10000");
             }
           };
 
       insertMany(rangeClosed(1, 50), coll);
       task.start(cfg);
 
-      List<SourceRecord> firstPoll = getNextResults(task);
```

Comment:
`getNextBatch` - polls and returns the first results it gets.
`getNextResults` - polls and combines results until there is no more data available.

This allows checking each poll call when required and also simplifies the general checking for data.